optuna/_gp/optim_mixed.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from optuna._gp.acqf import AcquisitionFunctionParams
+from optuna._gp.acqf import eval_acqf_no_grad
+from optuna._gp.acqf import eval_acqf_with_grad
+from optuna._gp.search_space import normalize_one_param
+from optuna._gp.search_space import sample_normalized_params
+from optuna._gp.search_space import ScaleType
+from optuna.logging import get_logger
+
+
+if TYPE_CHECKING:
+    import scipy.optimize as so
+else:
+    from optuna import _LazyImport
+
+    so = _LazyImport("scipy.optimize")
+
+_logger = get_logger(__name__)
+
+
+def _local_search_continuous(
+    acqf_params: AcquisitionFunctionParams,
+    initial_params: np.ndarray,
+    initial_fval: float,
+    continuous_params: np.ndarray,
+    continuous_param_scale: np.ndarray,
+    tol: float,
+) -> tuple[np.ndarray, float] | None:
+    if len(continuous_params) == 0:
+        return None
+    normalized_params = initial_params.copy()
+
+    def negfun_continuous_with_grad(x: np.ndarray) -> tuple[float, np.ndarray]:
+        normalized_params[continuous_params] = x * continuous_param_scale
```

Comment:
Isn't it actually the opposite?

```suggestion
        normalized_params[continuous_params] = x / continuous_param_scale
```

I guessed in this way because, e.g., Gaussian Kernel is:
$k(x, x') = \exp(- 1/2 * (x - x')^2 / l^2)$.
Then what we want after the transformation is:
$k(lx, lx') = \exp(- 1/2 * (lx - lx')^2 / l^2) = \exp(- 1/2 * (x - x')^2)$,
if my guess is correct.

Then the argument of this function should be $lx$ and `continuous_param_scale` takes $l$ if my check is correct.
In this case, we need to divide $x$ by $l$, i.e., `continuous_param_scale`.
If my statement is incorrect, please tell me!

Reply:
Your math is correct, and we want to make $x$ the parameter passed to scipy.optimize rather than $lx$ (which is the `normalized_params`).  So, we are multiplicating $x$ (the parameter given from scipy.optimize) with $l$ (lengthscale).

Reply:
Thank you for the comment!
Memo:
What we want to optimize is $h(x | k)$ and what `scipy.optimize` should optimize is $h(lz | k)$ w.r.t. $z$.